defold-spine/src/comp_spine_model.cpp

```diff
@@ -286,21 +295,62 @@ namespace dmSpine
 
         spAnimation* animation = spine_scene->m_Skeleton->animations[index];
 
-        component->m_AnimationId = animation_id;
-        component->m_AnimationInstance = spAnimationState_setAnimation(component->m_AnimationStateInstance, trackIndex, animation, loop);
+        if (track_index < 0)
+        {
+            dmLogError("Invalid track index %d", track_index);
+            return false;
+        }
 
-        component->m_Playback = playback;
-        component->m_AnimationInstance->timeScale = playback_rate;
-        component->m_AnimationInstance->reverse = IsReverse(playback);
-        component->m_AnimationInstance->mixDuration = blend_duration;
-        component->m_Playing = 1;
+        while (track_index >= component->m_AnimationTracks.Capacity())
+            component->m_AnimationTracks.SetCapacity(component->m_AnimationTracks.Capacity() * 2);
+        while (track_index >= component->m_AnimationTracks.Size())
+        {
+            SpineAnimationTrack track;
+            track.m_AnimationInstance = nullptr;
+            track.m_Playback = dmGameObject::PLAYBACK_NONE;
+            track.m_AnimationId = 0;
+            track.m_AnimationCallbackRef = 0;
+            dmMessage::ResetURL(&track.m_Listener);
```

Comment:
Some of these are needed:
* `m_AnimationId` is queried independently of `m_AnimationInstance` when getting the `animation` prop (should it now that I think about it? the current semantics of `animation` is "the last played animation" not "the currently playing animation if any")
* `m_AnimationCallbackRef` needs to be initialised because a few lines down the callback is unreffed in case there was a previous animation. `m_Listener` doesn't need to be initialised because `ClearCompletionCallback` clears it.

Reply:
In cases like this, I think it's good to have a function like `IsValid(Track)`.
I generally don't like having a state that is "half ok" (E.g. the animation id is ok, but the pointer is not)

But ofc, it's a case by case basis.